source/common/ur_util.hpp

```diff
@@ -196,23 +198,31 @@ inline std::optional<EnvVarMap> getenv_to_map(const char *env_var_name,
         std::stringstream kv_ss(key_value);
 
         if (reject_empty && key_value.find(':') == std::string::npos) {
-            throw_wrong_format_map(env_var_name);
+            throw_wrong_format_map(env_var_name, *env_var);
         }
 
         std::getline(kv_ss, key, key_value_delim);
         std::getline(kv_ss, values);
         if (key.empty() || (reject_empty && values.empty()) ||
-            values.find(':') != std::string::npos ||
             map.find(key) != map.end()) {
-            throw_wrong_format_map(env_var_name);
+            throw_wrong_format_map(env_var_name, *env_var);
         }
 
         std::vector<std::string> values_vec;
         std::stringstream values_ss(values);
         std::string value;
+        auto is_quoted = [&value]() {
+            return (value.front() == '\'' && value.back() == '\'') ||
+                   (value.front() == '"' && value.back() == '"');
+        };
         while (std::getline(values_ss, value, values_delim)) {
-            if (value.empty()) {
-                throw_wrong_format_map(env_var_name);
+            if (value.empty() ||
+                ((value.find(':') != std::string::npos) && !is_quoted())) {
```

Comment:
Since you have an anonymous function for `is_quoted` you might want to create `has_colon` for `(value.find(':') != std::string::npos)`.

Reply:
Added `has_colon()`. Also moved lambdas out of the `while` loop.